Make gene filtering optional in core workflow

cbethell opened the pull request:
**Issue Addressed**
Closes #303 

**What is the purpose of these changes?**
<!--Provide some background on the changes proposed.-->
Per issue #303, this PR makes the gene filtering step in the `core-analysis/01-filter-sce` optional by adding a `filter_genes` option and logic to the script.

**What changes did you make?**
<!--Describe the concrete changes that you made or additional features that were added, be as detailed as possible in the steps you took.-->
A `filter_genes` option was added to the `core-analysis/01-filter-sce.R` script, with a default of `FALSE`. Therefore, gene filtering will not be performed by default. If the `filter_genes` option _is_ provided, then the gene filtering threshold will be implemented for gene-level filtering.

The `README.md` file was also updated in this PR to include the added `filter_genes` parameter to the "filtering parameters" table.

**Any comments, concerns, or questions important for reviewers**
- Does this PR adequately address #303?

**Checklist**

Place an `x` in all boxes that you have completed.

- [x] I have run the most recent version of the code
- [ ] If I am adding in reports or generating any plots, I have reviewed the necessary reports
- [x] I have added all necessary documentation (if applicable)

allyhawkins commented on README.md:
```diff
@@ -251,6 +251,7 @@ Below are the parameters required to run either of the filtering methods.
 | `filtering_method` | `filtering_method`, the specified filtering method which can be one of "miQC" or "manual". For more information on choosing a filtering method, see [Filtering low quality cells](./additional-docs/processing-information.md#filtering-low-quality-cells) in the [processing information documentation](./additional-docs/processing-information.md) | "miQC" |
 | `prob_compromised_cutoff` | the maximum probability of a cell being compromised as calculated by [miQC](https://bioconductor.org/packages/release/bioc/html/miQC.html), which is required when the `filtering_method` is set to `miQC` in the project metadata | 0.75 |
+| `filter_genes` | a binary value indicating whether or not to perform gene fitering | `FALSE` |
 | `gene_detected_row_cutoff` | the percent of cells a gene must be detected in; genes detected are filtered regardless of the `filtering_method` specified in the project metadata | 5 |
```
```suggestion
| `gene_detected_row_cutoff` | the percent of cells a gene must be detected in; genes detected are only filtered if `filter_genes` is set to `TRUE` | 5 |
```
We should modify this statement for this parameter and the `gene_means_cutoff` below.

allyhawkins commented on core-analysis/01-filter-sce.R:
```diff
@@ -109,6 +110,12 @@ option_list <- list(
+    c("--filter_genes"),
+    action = "store_true",
+    default = FALSE,
+    help = "specifies whether or not to perform gene filtering"
```
```suggestion
    help = "specifies whether or not to perform gene filtering. Default is FALSE."
```

allyhawkins commented on core-analysis/01-filter-sce.R:
```diff
@@ -237,16 +244,18 @@ if (opt$filtering_method == "manual") {
+  detected <-
+    rowData(filtered_sce)$detected > opt$gene_detected_row_cutoff
+  expressed <- rowData(filtered_sce)$mean > opt$gene_means_cutoff
+  filtered_sce <- filtered_sce[detected & expressed,]
```
We should probably add a note in the metadata that genes are removed? And same for if genes are not removed?

allyhawkins reviewed (commented):
This generally looks pretty good. I would add a note in the metadata about gene filtering and if it was performed or not. 

Also can you update the [overview section](https://github.com/AlexsLemonade/scpca-downstream-analyses/tree/development#core-analysis-overview)? 
In the `1.Filtering` section you should note that removal of genes is optional.

cbethell commented:
Thanks for the review @allyhawkins -- I added a note to the metadata of the SCE object on whether or not genes have been filtered and updated the SCE content section of the `README`. I also made those additional doc updates you suggested, let me know what you think!

allyhawkins commented on core-analysis/01-filter-sce.R:
```diff
@@ -237,16 +244,21 @@ if (opt$filtering_method == "manual") {
+    rowData(filtered_sce)$detected > opt$gene_detected_row_cutoff
+  expressed <- rowData(filtered_sce)$mean > opt$gene_means_cutoff
+  filtered_sce <- filtered_sce[detected & expressed,]
+  metadata(filtered_sce)$genes_filtered <- "genes filtered"
+} else {
+  metadata(filtered_sce)$genes_filterd <- "no genes filtered"
```
```suggestion
} 
metadata(filtered_sce)$genes_filtered <- opt$filter_genes
```
I think we can actually just do this and set it to true/false which will be the value of the argument so we can do it on one line.

allyhawkins commented on core-analysis/01-filter-sce.R:
```diff
@@ -237,16 +244,21 @@ if (opt$filtering_method == "manual") {
+  metadata(filtered_sce)$genes_filtered <- "genes filtered"
+} else {
+  metadata(filtered_sce)$genes_filterd <- "no genes filtered"
+}
```
```suggestion
```
If you take my above suggestion.

allyhawkins reviewed (approved):
I just had one suggestion to simplify how we store the information in the metadata, but other than that looks good!
